apps/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -502,6 +502,9 @@ export abstract class BaseInstallManager {
         } else {
           args.push('--frozen-shrinkwrap');
         }
+      } else if (this._rushConfiguration.pnpmOptions.preferFrozenLockfileForUpdate) {
+        // In workspaces, we want to avoid unnecessary lockfile churn
```

Comment:
You may also want to condition this option on workspaces support, though opting in does imply that they know what they're doing

Reply:
If we condition it on workspaces, then the flag name would also need to imply that - something like preferFrozenLockfileForUpdateInWorkspaces. But like you said, opting in does imply that they know what they're doing, so IMO, this should not be conditioned on workspaces support.